src/lib.rs

```diff
@@ -72,25 +72,36 @@ impl Hasher {
     /// This works just like `Hasher::new`, except that it allows for an initial
     /// CRC32 state to be passed in.
     pub fn new_with_initial(init: u32) -> Self {
-        Self::internal_new_specialized(init).unwrap_or_else(|| Self::internal_new_baseline(init))
+        Self::internal_new_specialized(init, 0)
```

Comment:
Instead of duplicating the specialization/baseline constructor logic, could we make a call to the new constructor here, providing `0` as a length?